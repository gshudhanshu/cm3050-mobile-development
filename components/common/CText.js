import React from 'react'
import { Text, StyleSheet } from 'react-native'

const CText = ({ style, children, weight = 'regular', ...props }) => {
  let fontFamily
  switch (weight) {
    case 'light':
      fontFamily = 'Poppins_300Light'
      break
    case 'regular':
      fontFamily = 'Poppins_400Regular'
      break
    case 'medium':
      fontFamily = 'Poppins_500Medium'
      break
    case 'semiBold':
      fontFamily = 'Poppins_600SemiBold'
      break
    case 'bold':
      fontFamily = 'Poppins_700Bold'
      break
    default:
      fontFamily = 'Poppins_400Regular'
  }

  return (
    <Text style={[{ fontFamily }, style]} {...props}>
      {children}
    </Text>
  )
}

export default CText
