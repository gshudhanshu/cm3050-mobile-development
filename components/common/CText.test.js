import React from 'react'
import { render } from '@testing-library/react-native'
import CText from './CText'

describe('CText', () => {
  it('renders correctly', () => {
    const { getByText } = render(<CText>Test Text</CText>)
    expect(getByText('Test Text')).toBeTruthy()
  })

  it.each([
    ['light', 'Poppins_300Light'],
    ['regular', 'Poppins_400Regular'],
    ['medium', 'Poppins_500Medium'],
    ['semiBold', 'Poppins_600SemiBold'],
    ['bold', 'Poppins_700Bold'],
  ])(
    'applies the correct font family for weight: %s',
    (weight, expectedFontFamily) => {
      const { getByText } = render(<CText weight={weight}>Test Text</CText>)
      const textComponent = getByText('Test Text')
      expect(textComponent).toBeTruthy()
    }
  )

  it('applies custom styles', () => {
    const customStyle = { color: 'blue' }
    const { getByText } = render(<CText style={customStyle}>Test Text</CText>)
    const textComponent = getByText('Test Text')
    expect(textComponent).toBeTruthy()
  })
})
