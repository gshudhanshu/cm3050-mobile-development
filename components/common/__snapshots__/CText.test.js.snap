// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CText matches snapshot 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Poppins_400Regular",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
